lib/src/trailer.rs

```diff
@@ -0,0 +1,264 @@
+///
+/// Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+/// tempor incididunt ut labore et dolore magna aliqua.
+///
+/// Co-authored-by: Alice <alice@example.com>
+/// Co-authored-by: Bob <bob@example.com>
+/// Reviewed-by: Charlie <charlie@example.com>
+/// Change-Id: I1234567890abcdef1234567890abcdef12345678
+/// ```
+///
+/// In this case, there are four trailers: two `Co-authored-by` lines, one
+/// `Reviewed-by` line, and one `Change-Id` line.
+pub fn parse_description_trailers(body: &str) -> Vec<Trailer> {
+    let (trailer, blank) = parse_trailers_and_blank(body);
+    if blank {
+        trailer
+    } else {
+        // no blank found, this means there was a single paragraph, so whatever
+        // was found can't come from the trailer
+        vec![]
+    }
+}
+
+/// Parse the trailers from a trailer paragraph. This function behaves like
+/// `parse_description_trailer`, except that it doesn't expect the body to
+/// contain several paragraphs.
+pub fn parse_trailers(body: &str) -> Vec<Trailer> {
+    let (trailer, _) = parse_trailers_and_blank(body);
+    trailer
+}
+
+fn parse_trailers_and_blank(body: &str) -> (Vec<Trailer>, bool) {
+    // a trailer always comes at the end of a message; we can split the message
+    // by newline, but we need to immediately reverse the order of the lines
+    // to ensure we parse the trailer in an unambiguous manner; this avoids cases
+    // where a colon in the body of the message is mistaken for a trailer
+    let lines = body.trim_end().lines().rev();
+    let trailer_re =
+        regex::Regex::new(r"^([a-zA-Z0-9-]+) *: +(.+) *$").expect("trailer regex should be valid");
+    let mut trailer: Vec<Trailer> = Vec::new();
+    let mut multiline_value: Vec<&str> = Vec::new();
+    let mut found_blank = false;
+    for line in lines {
+        if line.starts_with(' ') {
+            multiline_value.push(line.trim());
+        } else if let Some(caps) = trailer_re.captures(line) {
+            let key = caps[1].trim().to_string();
+            multiline_value.push(caps.get(2).unwrap().as_str());
+            let value = multiline_value.iter().rev().join(" ");
+            multiline_value.clear();
+            trailer.push(Trailer { key, value });
+        } else if line.trim().is_empty() {
+            // end of the trailer
+            found_blank = true;
+            break;
+        } else {
+            // a non trailer in the trailer paragraph
+            // the line is ignored, as well as the multiline value that may
+            // have previously been accumulated
+            multiline_value.clear();
+        }
+    }
+    // reverse the insert order, since we parsed the trailer in reverse
+    trailer.reverse();
+    (trailer, found_blank)
+}
+
+#[cfg(test)]
+mod tests {
+    use pretty_assertions::assert_eq;
+
+    use super::*;
+
+    #[test]
+    fn test_simple_trailers() {
+        let body = r#"chore: update itertools to version 0.14.0
```

Comment:
nit: iiuc, the "body" is just the "lorem ipsum" part, so call these variables `description` instead?